pkg/instance/handler.go

```diff
@@ -76,6 +77,20 @@ func (h Handler) Deploy(c *gin.Context) {
 		}
 	}
 
+	preset := false
+	if presetParam, ok := c.GetQuery("preset"); ok {
+		var err error
+		if preset, err = strconv.ParseBool(presetParam); err != nil {
+			_ = c.Error(err)
+			return
+		}
+	}
+
+	if deploy && preset {
```

Comment:
`deploy` is set to `true` by default so this will if I read it correctly return an error even if the user did not set the query parameter `deploy=true`

The error should only be returned if query params are both `deploy=true` and `preset=true`.

The variables `deploy` and `preset` should never both be true. This makes the code further down the line easier IMHO.